knowledge-graph/src/test/scala/io/renku/knowledgegraph/projects/rest/GitLabProjectFinderSpec.scala

```diff
@@ -48,106 +53,90 @@ class GitLabProjectFinderSpec
-          .willReturn(
-            okJson(projectJson(project).noSpaces)
-          )
-      }
-
-      projectFinder.findProject(path).value.unsafeRunSync() shouldBe Some(project)
+      mapResponse(Status.Ok, Request[IO](), Response[IO]().withEntity(projectJson(project)))
+        .unsafeRunSync() shouldBe project.some
     }
 
     "return None if service responds with NOT_FOUND" in new TestCase {
-      implicit val accessToken: AccessToken = accessTokens.generateOne
-
-      val path = projectPaths.generateOne
-      stubFor {
-        get(s"/api/v4/projects/${urlEncode(path.toString)}?statistics=true")
-          .willReturn(notFound())
-      }
-
-      projectFinder.findProject(path).value.unsafeRunSync() shouldBe None
+      mapResponse(Status.NotFound, Request[IO](), Response[IO]()).unsafeRunSync() shouldBe None
     }
 
     "return a RuntimeException if remote client responds with status different than OK or NOT_FOUND" in new TestCase {
-      implicit val accessToken: AccessToken = accessTokens.generateOne
-
       val path = projectPaths.generateOne
-      stubFor {
-        get(s"/api/v4/projects/${urlEncode(path.toString)}?statistics=true")
-          .willReturn(unauthorized().withBody("some error"))
-      }
 
       intercept[Exception] {
-        projectFinder.findProject(path).value.unsafeRunSync()
-      }.getMessage shouldBe s"GET $gitLabUrl/api/v4/projects/${urlEncode(path.toString)}?statistics=true returned ${Status.Unauthorized}; body: some error"
+        mapResponse(Status.Unauthorized, Request[IO](), Response[IO]())
+      }
     }
 
     "return a RuntimeException if remote client responds with unexpected body" in new TestCase {
-      implicit val accessToken: AccessToken = accessTokens.generateOne
-
-      val path = projectPaths.generateOne
-      stubFor {
-        get(s"/api/v4/projects/${urlEncode(path.toString)}?statistics=true")
-          .willReturn(okJson("{}"))
-      }
 
       intercept[Exception] {
-        projectFinder.findProject(path).value.unsafeRunSync()
-      }.getMessage should startWith(
-        s"GET $gitLabUrl/api/v4/projects/${urlEncode(path.toString)}?statistics=true returned ${Status.Ok}; error: Invalid message body: Could not decode JSON: {}"
-      )
+        mapResponse((Status.Ok, Request[IO](), Response[IO]().withEntity(json"""{}"""))).unsafeRunSync()
+      }.getMessage should startWith(s"Invalid message body: Could not decode JSON:")
     }
   }
 
   private trait TestCase {
     implicit val logger: TestLogger[IO] = TestLogger[IO]()
-    val gitLabUrl     = GitLabUrl(externalServiceBaseUrl)
-    val projectFinder = new GitLabProjectFinderImpl[IO](gitLabUrl, Throttler.noThrottling)
+    val gitLabClient  = mock[GitLabClient[IO]]
+    val projectFinder = new GitLabProjectFinderImpl[IO](gitLabClient)
+
+    def uri(path: model.projects.Path) = uri"projects" / path.show withQueryParam ("statistics", "true")
+
+    val endpointName:         String Refined NonEmpty = "project"
```

Comment:
Maybe we could name then endpoint `single project` as that's more or less how it's named on the GitLab API docs